columnar/src/backend/columnar/columnar_reader.c

```diff
@@ -1823,11 +1830,25 @@ DeserializeChunkData(StripeBuffers *stripeBuffers, uint64 chunkIndex,
 				columnBuffers->chunkBuffersArray[chunkIndex];
 
 			/* decompress and deserialize current chunk's data */
-			StringInfo valueBuffer =
-				DecompressBuffer(chunkBuffers->valueBuffer,
+			StringInfo valueBuffer = NULL;
+			
+			if (columnar_enable_page_cache == true && chunkBuffers->valueCompressionType != COMPRESSION_NONE)
+			{
+				valueBuffer = ColumnarRetrieveCache(state->relation->rd_id, stripeId, chunkIndex, columnIndex);
+			}
+
+			if (valueBuffer == NULL)
+			{
+				valueBuffer = DecompressBuffer(chunkBuffers->valueBuffer,
```

Comment:
Can we decompress chunk into cache memory context first, so we don't copy data after into cache ?

Reply:
good point.